From: intrigeri <intrigeri@boum.org>
Date: Mon, 29 Jan 2018 08:19:04 +0000
Subject: AppArmor: support some of the included pluggable transports.

This fixes support for obfs4 and obfs3.

meek and fte require vastly more extended permissions and thus dedicated
child profiles.
---
 apparmor/torbrowser.Tor.tor | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/apparmor/torbrowser.Tor.tor b/apparmor/torbrowser.Tor.tor
index 7106a75..b0bfce0 100644
--- a/apparmor/torbrowser.Tor.tor
+++ b/apparmor/torbrowser.Tor.tor
@@ -16,11 +16,15 @@ profile torbrowser_tor @{torbrowser_tor_executable} {
   /etc/resolv.conf r,
   owner @{torbrowser_home_dir}/TorBrowser/Tor/tor mr,
   owner @{torbrowser_home_dir}/TorBrowser/Data/Tor/ rw,
-  owner @{torbrowser_home_dir}/TorBrowser/Data/Tor/* rw,
+  owner @{torbrowser_home_dir}/TorBrowser/Data/Tor/** rw,
   owner @{torbrowser_home_dir}/TorBrowser/Data/Tor/lock rwk,
   owner @{torbrowser_home_dir}/TorBrowser/Tor/*.so mr,
   owner @{torbrowser_home_dir}/TorBrowser/Tor/*.so.* mr,
 
+  # Support some of the included pluggable transports
+  owner @{torbrowser_home_dir}/TorBrowser/Tor/PluggableTransports/** rix,
+  @{PROC}/sys/net/core/somaxconn r,
+
   # Silence file_inherit logs
   deny @{torbrowser_home_dir}/{browser/,}omni.ja r,
   deny @{torbrowser_home_dir}/{browser/,}features/*.xpi r,
